python/GangaLHCb/Lib/RTHandlers/GaudiExecRTHandlers.py

```diff
@@ -330,10 +331,19 @@ def uploadLocalFile(job, namePattern, localDir, should_del=True):
     """
 
     new_df = DiracFile(namePattern, localDir=localDir)
-    random_SE = random.choice(getConfig('DIRAC')['allDiracSE'])
+    trySEs = getConfig('DIRAC')['allDiracSE']
+    random.shuffle(trySEs)
     new_lfn = os.path.join(getInputFileDir(job), namePattern)
-    returnable = new_df.put(force=True, uploadSE=random_SE, lfn=new_lfn)[0]
-
+    for SE in trySEs:
+        try:
+            returnable = new_df.put(force=True, uploadSE=SE, lfn=new_lfn)[0]
+        except:
```

Comment:
@mesmith75 Not sure that I understand the logic here. As it is, a `put` call raising an (unknown) will terminate the loop and not try the next SE. Is that the intention? Also, when we catch the exception (and subsequently raise the generic GangaException, could we forward some information about the original exception? At the moment it will be completely hidden what went wrong.